javalib/src/main/scala/java/util/StringTokenizer.scala

```diff
@@ -0,0 +1,106 @@
+/*
+ * Scala.js (https://www.scala-js.org/)
+ *
+ * Copyright EPFL.
+ *
+ * Licensed under Apache License 2.0
+ * (https://www.apache.org/licenses/LICENSE-2.0).
+ *
+ * See the NOTICE file distributed with this work for
+ * additional information regarding copyright ownership.
+ */
+
+package java.util
+
+class StringTokenizer(
+  str: String,
+  private var delim: String,
+  returnDelims: Boolean
+) extends java.util.Enumeration[Object] {
+
+  def this(str: String) = this(str, " \t\n\r\f", false)
+  def this(str: String, delim: String) = this(str, delim, false)
+
+  private var position: Int = 0
+  private val length: Int = str.length
+
+  def countTokens(): Int = {
+    var count: Int = 0
+    var inToken: Boolean = false
+    var i: Int = position
+
+    while(i < length) {
+      if (isDelim(str.charAt(i))) {
+        if (returnDelims)
+          count += 1
+
+        if (inToken) {
+          count += 1
+          inToken = false
+        }
+      } else {
+        inToken = true
+      }
+
+      i += 1
+    }
+
+    if (inToken)
+      count += 1
+
+    count
+  }
+
+  def hasMoreElements(): Boolean = hasMoreTokens()
+
+  def hasMoreTokens(): Boolean = {
+    if (position >= length) false
+    else if (returnDelims) true
+    else {
```

Comment:
This may seem silly sometimes, but our coding style rule of having all the branches and if/else's agree on whether to use braces or not means that this should be formatted as
```suggestion
    if (position >= length) {
      false
    } else if (returnDelims) {
      true
    } else {
```